index.js

```diff
@@ -1,11 +1,28 @@
 var N = require('libnested')
 
 var isModule = require('./is')
+var apply = require('./apply')
+
+module.exports = function combine () {
+  var modules = flattenNested(arguments)
+
+  throwIfMissingDeps(modules)
 
-function hasAll(set, keys) {
-  return keys.every(function (k) {
-    return !!set[k]
+  var api = {}
+
+  eachModule(modules, function (module, key) {
+    var deps = buildDeps(module.needs, api)
+    var exported = module.create(deps)
+
+    if (!exported) { throw new Error('export declared but not returned for', key) }
```

Comment:
i think you need to concat your error message, as the `Error` constructor only accepts one arg:

```js
> new Error('hi', 'there')
Error: hi
```

@mixmix `key` is the name of the module definition, which can give, need, and create many modules each with a `path`. this checks that the create returned something (which corresponds to the module definition `key`), not that it returned each module (which corresponds to the module `path`).